File: backend/IMIS/IMIS.Application/KraRoadMapModule/KraRoadMapFilter.cs
namespace IMIS.Application.KraRoadMapModule
{
    public class KraRoadMapFilter
    {
        public long Id { get; set; }
        public int KraId { get; set; }
        public long KraRoadMapId { get; set; }
        public string? KraDescription { get; set; }
        public int Year { get; set; }
        public string? DeliverableDescription { get; set; }      
    }
}

File: frontend/imis/lib/performance_governance_system/pgs_roadmap/models/kra_roadmap_filter.dart
import 'package:json_annotation/json_annotation.dart';
part 'kra_roadmap_filter.g.dart';

@JsonSerializable()
class KraRoadmapFilter {
  int? id;
  int kraId;
  int? kraRoadMapId;
  String kraDescription;
  int year;
  String? deliverableDescription;
  bool isDirect;

  KraRoadmapFilter({
    this.id,
    required this.kraId,
    this.kraRoadMapId,
    required this.kraDescription,
    required this.year,
    this.deliverableDescription,
    required this.isDirect,
  });

  factory KraRoadmapFilter.fromJson(Map<String, dynamic> json) =>
      _$KraRoadmapFilterFromJson(json);

  Map<String, dynamic> toJson() => _$KraRoadmapFilterToJson(this);
}
